js/books.js

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable max-classes-per-file */
+class StorageUtil {
+  localStorageAvailable() {
+    this.storage = null;
+
+    try {
+      this.storage = window.localStorage;
+      const x = '__storage_test__';
+      this.storage.setItem(x, x);
+      this.storage.removeItem(x);
+      return true;
+    } catch (e) {
+      return e instanceof DOMException && (
+        // everything except Firefox
+        e.code === 22
+        // Firefox
+        || e.code === 1014
+        // test name field too, because code might not be present
+        // everything except Firefox
+        || e.name === 'QuotaExceededError'
+        // Firefox
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+        // acknowledge QuotaExceededError only if there's something already stored
+        && (this.storage && this.storage.length !== 0);
+    }
+  }
+
+  setItem(name, item) {
+    if (this.localStorageAvailable()) {
+      localStorage.setItem(name, item);
+    }
+  }
+
+  getItem(name) {
+    if (this.localStorageAvailable()) {
+      return localStorage.getItem(name);
+    }
+    return null;
+  }
+}
+
+class Book {
+  constructor(id, title, author) {
+    this.id = id;
+    this.title = title;
+    this.author = author;
+  }
+}
+
+class Library {
+  constructor() {
+    this.books = [];
+    this.storageUtil = new StorageUtil();
+    this.init();
+  }
+
+  init() {
+    this.books = JSON.parse(this.storageUtil.getItem('books')) || [];
+  }
+
+  save() {
+    this.storageUtil.setItem('books', JSON.stringify(this.books));
+  }
+
+  addBook(title, author) {
+    const id = this.books.length + 1;
+    const book = new Book(id, title, author);
+
+    this.books.push(book);
```

Comment:
- [ ] **[OPTIONAL]** It's best practice to avoid array mutation. Instead of using `push`, you may use methods that do not mutate the array. Try `concat`, `filter`, `slice`.